Pre-release

Bobby-Huggins opened the pull request:
1. Adds `save()` methods and `load()` class methods to `VBMC` and `VariationalPosterior`, adds tests for the same, and updates the Jupyter Notebook examples to show these interfaces.
2. Fixes a small issue with `VariationalPosterior.plot()` that prevented GP samples from being plotted, and adds some simple options to `create_vbmc_animation()`.
3. Adds links to PyBADS to the documentation, and removes superfluous links to MATLAB versions of VBMC/BADS.
4. Updates the instructions on building documentation in _Instructions for developers and contributors_ to explain that documentation is built automatically (but should still be checked locally as necessary).

lacerbi commented:
The changes are okay but it looks like some tests are failing, so I'd wait until we clarify that.

Bobby-Huggins commented:
Assuming the latest round of tests passes, this should be ready to merge. As we discussed, I removed the explicit version-dependence of the tests, and added a note to the docstring and to [Example 3](https://github.com/acerbilab/pyvbmc/blob/pre-release/examples/pyvbmc_example_3_diagnostics_and_saving.ipynb) that pickling + unpickling functions across Python versions can be unreliable.

Unrelated, but I also restructured the `VBMC` and `VariationalPosterior` `.rst` files so that links to the most important user-facing methods appear in the right-hand-side navigation panel (e.g. `VBMC.optimize()`, `VBMC.save()`, `VBMC.load()`; `VariationalPosterior.sample()` + several more).
